### PR TITLE
DBAAS-1191: parse v1alpha1 provider instance spec and status

### PR DESCRIPTION
**Pull request by xieshenzh**
## Description

Parse v1alpha1 provider instance spec and status.
Jira: https://issues.redhat.com/browse/DBAAS-1191

## Verification Steps

1. Create RDS Inventory
2. Create RDS Instance

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: api/v1alpha1/dbaasinstance_conversion.go [hunk 1]
```diff
@@ -130,3 +130,19 @@ func (dst *DBaaSInstanceSpec) ConvertFrom(src *v1beta1.DBaaSInstanceSpec) error
 	}
 	return nil
 }
+
+// ConvertTo convert this DBaaSInstanceStatus to v1beta1.DBaaSInstanceStatus
+func (src *DBaaSInstanceStatus) ConvertTo(dst *v1beta1.DBaaSInstanceStatus) {
```
Comment (tchughesiv) [inline api/v1alpha1/dbaasinstance_conversion.go:135]: same, but from main `ConvertTo` function?

File: api/v1alpha1/dbaasinstance_conversion.go [hunk 2]
```diff
@@ -130,3 +130,19 @@ func (dst *DBaaSInstanceSpec) ConvertFrom(src *v1beta1.DBaaSInstanceSpec) error
...
+	dst.Conditions = src.Conditions
+	dst.InstanceID = src.InstanceID
+	dst.InstanceInfo = src.InstanceInfo
+	dst.Phase = v1beta1.DBaasInstancePhase(src.Phase)
+}
+
+// ConvertFrom converts the DBaaSInstanceStatus from the v1beta1 to this version.
+func (dst *DBaaSInstanceStatus) ConvertFrom(src *v1beta1.DBaaSInstanceStatus) {
```
Comment (tchughesiv) [inline api/v1alpha1/dbaasinstance_conversion.go:143]: shouldn't this now be referenced from this function... within this same file?
```go
func (dst *DBaaSInstance) ConvertFrom(srcRaw conversion.Hub) error {
```

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: api/v1alpha1/dbaasinstance_conversion.go (hunk 2 above)
Comment (xieshenzh) [reply api/v1alpha1/dbaasinstance_conversion.go:137]: Sure, thanks.

**Review by tchughesiv — APPROVED** [review]
lgtm

**Review by jeremyary — APPROVED** [review]

**Pull request merged**